pkg/pool-utils/contracts/BasePool.sol

```diff
@@ -30,15 +30,11 @@ import "@balancer-labs/v2-asset-manager-utils/contracts/IAssetManager.sol";
 import "./BalancerPoolToken.sol";
 import "./BasePoolAuthorization.sol";
 
-// This contract relies on tons of immutable state variables to perform efficient lookup, without resorting to storage
-// reads. Because immutable arrays are not supported, we instead declare a fixed set of state variables plus a total
-// count, resulting in a large number of state variables.
-
 // solhint-disable max-states-count
 
 /**
- * @dev Reference implementation for the base layer of a Pool contract that manages a single Pool with an immutable set
- * of registered tokens, no Asset Managers, an admin-controlled swap fee percentage, and an emergency pause mechanism.
+ * @dev Reference implementation for the base layer of a Pool contract that manages a single Pool with no
+ * Asset Managers, an admin-controlled swap fee percentage, and an emergency pause mechanism.
```

Comment:
```suggestion
 * @dev Reference implementation for the base layer of a Pool contract that manages a single Pool with optional
 * Asset Managers, an admin-controlled swap fee percentage, and an emergency pause mechanism.
```

This comment is stale as we allow these now.

Reply:
Missed that; thanks. And as discussed, I'll add a factory that hard-codes zeroes so that we can whitelist a factory without having to worry about people sneaking in rogue asset managers